model/shared/src/main/scala/explore/model/ProgramSummaries.scala

```diff
@@ -212,6 +213,57 @@ case class ProgramSummaries(
       .toMap
   end groupsChildren
 
+  private def allObservationsForGroup(groupId: Group.Id): List[Observation] =
```

Comment:
I don't know if we need to worry about stack safety here or not. I'm not sure if the ODB enforces a limit to the depth of the group tree, but:
1. It seems unlikely someone will build a tree hundreds of groups deep, and
2. If they do, it will probably bring the scheduler to it's knees and we'll need to implement a limit.

Reply:
also would kill time calculation

Reply:
I decided to limit the recursion level just in case. It will only affect warning generation.